rust-runtime/aws-smithy-types/src/date_time/format.rs

```diff
@@ -377,6 +377,12 @@ pub(crate) mod rfc3339 {
     // Timezones not supported:
     // Not OK: 1985-04-12T23:20:50-02:00
     pub(crate) fn parse(s: &str) -> Result<DateTime, DateTimeParseError> {
+        let timezone_offset = "-02:00".len() - 1;
+        if matches!(s.chars().rev().nth(timezone_offset), Some('-') | Some('+')) {
+            return Err(DateTimeParseError::Invalid(
+                "invalid RFC-3339 date-time: timezone offset not supported".into(),
```

Comment:
This will break the SDK since the `HttpCredentialProvider` uses this module to parse the credential expiration time, and the SDK JSON credentials format allows for timezone offsets. You can see the call here: https://github.com/awslabs/smithy-rs/blob/ea2ae7bfc6d8ba1daf7a4648a4027b7268f56283/aws/rust-runtime/aws-config/src/json_credentials.rs#L181

I'm happy our unit tests caught this 😄 

I don't have any good suggestions at the moment for working around this. @rcoh or @Velfi might have some ideas. It's probably a tradeoff between modifying the `aws-smithy-types` public API or using the `time` crate directly for parsing in `aws-config`.

Reply:
I'd expect things in this crate to follow the idiosyncrasies of the Smithy spec. It appears like the Rust docs already mention that this wasn't intended to be fully compliant for RFC3339 by calling out that timezones are not supported.

Reply:
let's use `time` directly in aws-config and keep smithy-types smithy compliant. It was just a convenience that I used Smithy to do it—I didn't realize that was slightly wrong